kratos/sources/kratos_filesystem.cpp

```diff
@@ -132,5 +133,23 @@ void MPISafeCreateDirectories(const std::string& rPath)
     }
 }
 
+
+std::string ResolveSymlinks(const std::string& rPath)
+{
+    std::filesystem::path path = rPath;
+    KRATOS_ERROR_IF_NOT(std::filesystem::exists(path)) << "File not found: " << rPath;
+
+    std::set<std::filesystem::path> symlinks;
```

Comment:
why not unordered?

Reply:
I don't expect many items in that set (probably in the single digits) so a hash table would be an overkill. In fact, I have a hard time imagining a scenario in which more than 2-3 indirections would be reasonable. A couple of lexicographical string comparisons in ```std::set``` surely outperform computing the hash of a lengthy absolute path.